ios/Exponent/Kernel/Services/Notifications/EXRemoteNotificationManager.m

```diff
@@ -208,26 +194,31 @@ - (void)_synchronizeCurrentAPNSToken:(NSData *)currentToken
   }];
 }
 
-- (BOOL)_canRegisterForRemoteNotifications
+- (void)_canRegisterForRemoteNotificationsWithHandler:(void (^)(BOOL can))handler
 {
   dispatch_assert_queue(_queue);
-  
+
   // When the user has not granted permission to display any type of notification, iOS doesn't
   // invoke the delegate methods and registering for remote notifications will never complete
   //
   // TODO: Switch this to use UNNotificationSettings for iOS 10+. Note that UNNotificationCenter
   // is not thread-safe but can run off of the main thread (ex: a serial queue for notifications).
   dispatch_assert_queue(dispatch_get_main_queue());
-  UIUserNotificationSettings * settings = RCTSharedApplication().currentUserNotificationSettings;
-  return settings.types != UIUserNotificationTypeNone;
+  [[UNUserNotificationCenter currentNotificationCenter] getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings * _Nonnull settings) {
+    if (settings.authorizationStatus == UNAuthorizationStatusAuthorized) {
```

Comment:
`handler(settings.authorizationStatus == UNAuthorizationStatusAuthorized)` would be a bit more concise 🙂